core/repository/sparql/src/main/java/org/eclipse/rdf4j/repository/sparql/SPARQLConnection.java

```diff
@@ -405,6 +404,11 @@ public TupleQuery prepareTupleQuery(QueryLanguage ql, String query, String base)
 		throw new UnsupportedQueryLanguageException("Unsupported query language " + ql);
 	}
 
+	@Override
+	public void prepare() throws RepositoryException {
+		// no-op, not supported in SPARQL protocol
```

Comment:
I would prefer an unsupported operation exception, so users don’t spend time trying to figure out why prepare isn’t doing anything.

Reply:
Yeah good point, I'll do the same for the default method.